baybe/strategies/base.py

```diff
@@ -3,26 +3,48 @@
 from abc import ABC, abstractmethod
 from typing import Optional
 
+import cattrs
 import pandas as pd
 from attrs import define, field
 
-from baybe.recommenders.base import Recommender
+from baybe.exceptions import DeprecationError
+from baybe.recommenders.base import Recommender, RecommenderProtocol
+from baybe.recommenders.deprecation import structure_recommender_protocol
 from baybe.searchspace import SearchSpace
 from baybe.serialization import SerialMixin, converter, unstructure_base
-from baybe.strategies.deprecation import structure_strategy
 
 
 @define
-class Strategy(SerialMixin, ABC):
+class Strategy(SerialMixin, RecommenderProtocol, ABC):
     """Abstract base class for all BayBE strategies."""
 
-    allow_repeated_recommendations: bool = field(default=False, kw_only=True)
-    """Allow to make recommendations that were already recommended earlier. This only
-    has an influence in discrete search spaces."""
+    allow_repeated_recommendations: bool = field(default=None, kw_only=True)
+    """Deprecated! The flag has become an attribute of
+    :class:`baybe.recommenders.base.Recommender`."""
 
-    allow_recommending_already_measured: bool = field(default=False, kw_only=True)
-    """Allow to output recommendations that were measured previously. This only has an
-    influence in discrete search spaces."""
+    allow_recommending_already_measured: bool = field(default=None, kw_only=True)
+    """Deprecated! The flag has become an attribute of
+    :class:`baybe.recommenders.base.Recommender`."""
+
+    @allow_repeated_recommendations.validator
+    def validate_allow_repeated_recommendations(self, _, value):
```

Comment:
Should this (as well as the other corresponding functions) be public? This will make them appear in the documentation, so we should be sure that we want this.

Reply:
good catch, fixed